### PR TITLE
Update data generation to support API models

### PR DESCRIPTION
**Pull request by DylanLi-Hang**
Update implementation of data generation of [Improve Mathematical Reasoning in Language Models by Automated Process Supervision](https://arxiv.org/pdf/2406.06592) to support API models.

### CODE REVIEW

File: data/README.md [hunk 1]
```diff
@@ -4,6 +4,27 @@ A simple implementation of data generation of [Improve Mathematical Reasoning in
...
+
+processing:
+  initial_rollouts: 20
+  num_rollouts: 5
+
+model:
+  model_type: "hf"  # Supported types: "hf", "openai", "anthropic"
+  model_name: "Qwen/Qwen2.5-Math-7B-Instruct" # model name, e.g. "Qwen/Qwen2.5-Math-7B-Instruct", "gpt-4o"
+  model_args:
+    max_tokens: 512
```
Comment (ziyuwan) [inline data/README.md:24]: is `max_tokens` 512 sufficiently large?

Comment (iamlilAJ) [reply data/README.md:24]: Hi, thank you for your contribution! We recently reimplemented [OmegaPRM](https://github.com/openreasoner/openr/tree/main/data/omegaPRM_v2), and I believe your code for supporting API models could integrate smoothly with our current setup. Let us know if you'd like to collaborate further on this! 
Thanks again!

**Comment by iamlilAJ** [comment]
Hi, thank you for your contribution! We recently reimplemented [OmegaPRM](https://github.com/openreasoner/openr/tree/main/data/omegaPRM_v2), and I believe your code for supporting API models could integrate smoothly with our current setup. Let us know if you'd like to collaborate further on this! 
Thanks again!

**Comment by DylanLi-Hang** [comment]
> Hi, thank you for your contribution! We recently reimplemented [OmegaPRM](https://github.com/openreasoner/openr/tree/main/data/omegaPRM_v2), and I believe your code for supporting API models could integrate smoothly with our current setup. Let us know if you'd like to collaborate further on this! Thanks again!

You're very welcome, I’d be happy to collaborate further on it!

File: data/README.md (hunk 1 above)
Comment (DylanLi-Hang) [reply data/README.md:24]: > is `max_tokens` 512 sufficiently large?

A `max_tokens` setting of 512 might not be enough for complex or long responses, but we can change this setting in the config to fit different needs.

**Comment by mengfn** [comment]
Thank you for implementing this.

**Pull request merged**